e2e/journey-app/components/device-profile.ts

```diff
@@ -19,14 +24,40 @@ export default function deviceProfileComponent(
 
   journeyEl?.appendChild(message);
 
-  // Device profile callback typically runs automatically
-  // The callback will collect device information in the background
-  setTimeout(() => {
+  // Automatically trigger device profile collection
+  setTimeout(async () => {
     try {
-      // Device profile collection is typically handled automatically by the callback
-      console.log('Device profile collection initiated');
+      const isLocationRequired = callback.isLocationRequired();
+      const isMetadataRequired = callback.isMetadataRequired();
+
+      console.log('Collecting device profile...', { isLocationRequired, isMetadataRequired });
+
+      // Create device instance and collect profile
+      const device = new Device();
+      const profile = await device.getProfile({
+        location: isLocationRequired,
+        metadata: isMetadataRequired,
+      });
+
+      console.log('Device profile collected successfully');
+
+      // Set the profile on the callback
+      callback.setProfile(profile);
+      message.innerText = 'Device profile collected successfully!';
+      message.style.color = 'green';
+
+      // Auto-submit the form after successful collection
+      setTimeout(() => {
+        const submitButton = document.getElementById('submitButton') as HTMLButtonElement;
+        if (submitButton) {
+          submitButton.click();
+        }
```

Comment:
Do we not have a way to call the submit function without having to create a button and auto-clicking it?